static/index.js

```diff
@@ -91,4 +91,50 @@ document.addEventListener("DOMContentLoaded", event => {
     }
   });
 
+  var darkSwitch = document.querySelector('input[name=darkMode]');
+  const prefersDarkScheme = window.matchMedia("(prefers-color-scheme: dark)");
+  const prefersLightScheme = window.matchMedia("(prefers-color-scheme: light)");
+  const _darkModeLocal = localStorage.getItem("darkModeLocal");
+
+  darkSwitch.addEventListener('change', function (_event) {
+    // toggles darkmode when user taps switch
+    if (localStorage.getItem("darkModeLocal") == "dark") {
+      document.documentElement.classList.toggle("dark", false);
+      document.getElementById("darkMode").checked = false;
+      localStorage.setItem("darkModeLocal", "light");
+    }
+    else if (localStorage.getItem("darkModeLocal") == "light") {
+      document.documentElement.classList.toggle("dark", true);
+      document.getElementById("darkMode").checked = true;
+      localStorage.setItem("darkModeLocal", "dark");
+    }
+  });
+
+  function configureDisplayTheme() {
+    // if first time, then set theme based on system preference
+    if (_darkModeLocal === null) {
+      if (prefersDarkScheme.matches) {
+        localStorage.setItem("darkModeLocal", "dark");
```

Comment:
Since we are repeating some code here, we can use the first condition to set localStorage if it's null.
```js
if (localStorage.get(appThemeLocalKey) == null) {
    // set local storage based on user preference
}
// now use the local storage as source of truth
if (localStorage.get(appThemeLocalKey) == "dark") {
    // change toggle and class
} else { // since it's set, there should be no third case - either dark or light 
   // change toggle and class
}
```